dcrec/secp256k1/curve.go

```diff
@@ -243,10 +275,10 @@ func addZ1EqualsZ2(p1, p2, result *JacobianPoint) {
 	e.Mul2(x1, &b)                         // E = X1*B (mag: 1)
 	negE.Set(&e).Negate(1)                 // negE = -E (mag: 2)
 	f.Mul2(x2, &b)                         // F = X2*B (mag: 1)
-	x3.Add2(&e, &f).Negate(3).Add(&d)      // X3 = D-E-F (mag: 5)
-	negX3.Set(x3).Negate(5).Normalize()    // negX3 = -X3 (mag: 1)
-	y3.Set(y1).Mul(f.Add(&negE)).Negate(3) // Y3 = -(Y1*(F-E)) (mag: 4)
-	y3.Add(e.Add(&negX3).Mul(&c))          // Y3 = C*(E-X3)+Y3 (mag: 5)
+	x3.Add2(&e, &f).Negate(2).Add(&d)      // X3 = D-E-F (mag: 4)
+	negX3.Set(x3).Negate(4)                // negX3 = -X3 (mag: 5)
+	y3.Set(y1).Mul(f.Add(&negE)).Negate(1) // Y3 = -(Y1*(F-E)) (mag: 2)
+	y3.Add(e.Add(&negX3).Mul(&c))          // Y3 = C*(E-X3)+Y3 (mag: 3)
 	z3.Mul2(z1, &a)                        // Z3 = Z1*A (mag: 1)
```

Comment:
Having a magnitude of 1 is not the same thing as being normalized.

The comment on `Mul2` calls out:

```Go
// Output Normalized: No
// Output Max Magnitude: 1
func (f *FieldVal) Mul2(val *FieldVal, val2 *FieldVal) *FieldVal {
```

So, it does still need to be normalized.

Reply:
Ah, got it.  Thanks for clarifying.  The comment below that states `Normalize the resulting field values to a magnitude of 1 as needed` threw me off.  I also noticed that there must not be a test hitting that case since if I comment out the `z3.Normalize()`, all of the tests still pass.

Reply:
Good point.  That comment does seems to imply otherwise and not what it really means.  I'll give things a pass regarding this to make it more clear.

Reply:
Oh, I didn't comment about your last bit regarding the tests still passing.

That's because the normalizations are not strictly needed for correctness given the specific way they are used in the existing implementation code because there are enough normalizations elsewhere in the calculations to avoid internal overflows.

However, there is no guarantee on exactly how the exported funcs will be used and their APIs specifically state that:

```Go
// NOTE: The points must be normalized for this function to return the correct
// result.  The resulting point will be normalized.
func AddNonConst(p1, p2, result *JacobianPoint) {
```

Therefore, they do need to be normalized to satisfy the API contract.

Reply:
That makes sense, thanks for the explanation!

Reply:
I updated the comments (as a part of the comment update commit PR this one depends on) to remove the part about a magnitude of 1 since it was misleading as pointed out.